src/libsystemd-network/sd-dhcp-lease.c

```diff
@@ -1030,21 +1031,21 @@ int dhcp_lease_save(sd_dhcp_lease *lease, const char *lease_file) {
         r = sd_dhcp_lease_get_dns(lease, &addresses);
         if (r > 0) {
                 fputs("DNS=", f);
-                serialize_in_addrs(f, addresses, r, false, NULL);
+                serialize_in_addrs(f, addresses, r, &wls, NULL);
```

Comment:
so this is wrong, should be NULL. otherwise we'll end up outputting a leading whitespace here if there's a ROUTER= field set